### PR TITLE
[Refactoring] Break deterministic mns circular dependencies

### PR DESCRIPTION
**Pull request by random-zebra**
This PR removes several circular dependencies related to deterministic masternodes code.

The chain of dependencies now is:
`validation` -> `specialtx` (containing the specialtx validation/processing code) -> `deterministicmns` (containing the dmn manager)

Functions defined in `specialtx` now take a view of the coins cache as parameter, thus no longer depend on `validation` and only require `coins`.
Both `specialtx` and `deterministicmns` depend on `providertx`, which now is responsible only for the definition of each payload class and does not depend on any of the previous modules.

Also, add more unit tests for invalid collateral in ProReg txes.

### CODE REVIEW

File: src/test/evo_deterministicmns_tests.cpp [hunk 1]
```diff
@@ -344,6 +345,7 @@ BOOST_FIXTURE_TEST_CASE(dip3_protx, TestChain400Setup)
         BOOST_CHECK_EQUAL(chainTip->nHeight, nHeight + 1);
         SyncWithValidationInterfaceQueue();
         BOOST_CHECK(deterministicMNManager->GetListAtChainTip().HasMN(txid));
+        view = pcoinsTip.get();
```
Comment (furszy) [inline src/test/evo_deterministicmns_tests.cpp:354]: why so many view pointer updates? none of them should be needed.

**Review by furszy — APPROVED** [review]
Code review ACK f7c2663.

extra nit: would rename the `specialtx.h/cpp` files to `validation_specialtx.h/cpp`.

File: src/test/evo_deterministicmns_tests.cpp (hunk 1 above)
Comment (random-zebra) [reply src/test/evo_deterministicmns_tests.cpp:354]: Yeah, not needed :+1:

**Comment by random-zebra** [comment]
Rebased, file renamed, and dropped unused header imports.
Also removed unneded view-pointer updates, and added "collateral-spent" case in the unit test (squashed in d04cf0f655d6d6422034d63089b596c4b637b110).

**Review by furszy — DISMISSED** [review]
ACK cbc874e

**Comment by random-zebra** [comment]
Rebased, fixing conflict in the circular dependency list.

**Review by furszy — DISMISSED** [review]
re-ACK 0a67b8f after rebase.

**Review by furszy — APPROVED** [review]
rebase utACK 037abb5200f785996bdc9b6473eb38dac3446e13

**Review by Fuzzbawls — APPROVED** [review]
ACK 037abb5200f785996bdc9b6473eb38dac3446e13

**Pull request merged**